instrumentation/kafka-streams/src/main/java/brave/kafka/streams/KafkaStreamsTracing.java

```diff
@@ -1,23 +1,105 @@
 package brave.kafka.streams;
 
+import brave.Span;
+import brave.SpanCustomizer;
+import brave.Tracing;
 import brave.kafka.clients.KafkaTracing;
+import brave.propagation.TraceContext;
+import brave.propagation.TraceContextOrSamplingFlags;
+import java.util.Properties;
+import org.apache.kafka.common.header.Headers;
 import org.apache.kafka.streams.KafkaClientSupplier;
+import org.apache.kafka.streams.KafkaStreams;
+import org.apache.kafka.streams.Topology;
+import org.apache.kafka.streams.kstream.Transformer;
+import org.apache.kafka.streams.kstream.TransformerSupplier;
+import org.apache.kafka.streams.kstream.ValueTransformer;
+import org.apache.kafka.streams.kstream.ValueTransformerSupplier;
+import org.apache.kafka.streams.kstream.ValueTransformerWithKey;
+import org.apache.kafka.streams.kstream.ValueTransformerWithKeySupplier;
+import org.apache.kafka.streams.processor.Processor;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.ProcessorSupplier;
 
-/** Use this class to decorate your Kafka Streams client. */
+/** Use this class to decorate Kafka Stream Topologies and enable Tracing. */
 public final class KafkaStreamsTracing {
 
-  public static KafkaStreamsTracing create(KafkaTracing kafkaTracing) {
-    return new KafkaStreamsTracing(kafkaTracing);
+  final Tracing tracing;
+  final TraceContext.Extractor<Headers> extractor;
+
+  KafkaStreamsTracing(Builder builder) { // intentionally hidden constructor
+    this.tracing = builder.tracing;
+    this.extractor = tracing.propagation().extractor(KafkaStreamsPropagation.GETTER);
+  }
+
+  public static KafkaStreamsTracing create(Tracing tracing) {
+    return new KafkaStreamsTracing.Builder(tracing).build();
+  }
+
+  static void addTags(ProcessorContext processorContext, SpanCustomizer result) {
+    result.tag(KafkaStreamsTags.KAFKA_STREAMS_APPLICATION_ID_TAG, processorContext.applicationId());
+    result.tag(KafkaStreamsTags.KAFKA_STREAMS_TASK_ID_TAG, processorContext.taskId().toString());
+  }
+
+  /**
+   * Creates a {@link KafkaStreams} instance with a tracing-enabled {@link KafkaClientSupplier}. All
+   * Topology Sources and Sinks (including internal Topics) will create a Spans on records
+   * processed.
+   */
+  public KafkaStreams kafkaStreams(Topology topology, Properties streamsConfig) {
+    return new KafkaStreams(topology, streamsConfig, kafkaClientSupplier());
+  }
+
+  /**
+   * Create a tracing-decorated {@link ProcessorSupplier}
+   */
+  public <K, V> ProcessorSupplier<K, V> processor(String name, Processor<K, V> processor) {
+    return new TracingProcessorSupplier<>(this, name, processor);
   }
 
-  final KafkaTracing kafkaTracing;
+  /**
+   * Create a tracing-decorated {@link TransformerSupplier}
+   */
+  public <K, V, R> TransformerSupplier<K, V, R> transformer(String name,
```

Comment:
will help to add javadoc code example, nicely formatted showing example usage.

While not relevant, here is the style I'm hoping for on these public methods.

```java
  /**
   * Use this to temporarily override the sampler used when starting new traces. This also serves
   * advanced scenarios, such as {@link DeclarativeSampler declarative sampling}.
   *
   * <p>Simple example:
   * <pre>{@code
   * // Ensures new traces are always started
   * Tracer tracer = tracing.tracer().withSampler(Sampler.ALWAYS_SAMPLE);
   * }</pre>
   *
   * @see DeclarativeSampler
   */
```

Reply:
@adriancole I've added comments with samples for all public API methods, wdyt?